src/core/server/http/base_path_proxy_server.ts

```diff
@@ -143,12 +143,25 @@ export class BasePathProxyServer {
       handler: {
         proxy: {
           agent: this.httpsAgent,
-          host: this.server.info.host,
           passThrough: true,
-          port: this.devConfig.basePathProxyTargetPort,
-          // typings mismatch. h2o2 doesn't support "socket"
-          protocol: this.server.info.protocol as HapiProxy.ProxyHandlerOptions['protocol'],
           xforward: true,
+          mapUri: async (request) => {
+            return {
+              // Passing in the headers is a workaround until this is fixed:
+              // https://github.com/hapijs/h2o2/issues/124
+              headers:
```

Comment:
Could you add this logic to the proxy below? https://github.com/elastic/kibana/pull/87270/files#diff-efcad692e769a93e63c54b04526c48cde9f6fc84dfd9e2781b679e440300b649R190

Reply:
Yeah I can add it.

Reply:
Actually if you're talking about the proxy right below this one it already has this:

```ts
headers: request.headers,
```

Which will correctly merge in the content-length correctly